### PR TITLE
actually take a slice in this example

### LINKED ISSUE #45678: [rustdoc]A wrong in documents about slice
https://doc.rust-lang.org/std/primitive.slice.html
I think that in `let x = &mut [1, 2, 3];`, `x` is not slice. It is just a reference for primitive type (arrary)
for example:

```
let intarr_slice = &[&[1,2][..],&[3,4,5][..]];
let intarr_slice2 = &[&[1,2],&[3,4,5]];
```
that the second on is not a right statements.
against the definition of `let str_slice: &[&str] = &["one", "two", "three"];`
`x = &mut[1,2,3][..]` is slice type.

And I think it's better to add a note.
```
let x = [1,2,3];
let y = &x[..];
```
and
```
let x = &[1,2,3];
let y = &x[..];
```
these y are same.

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #45678

### CODE REVIEW

File: src/libstd/primitive_docs.rs [hunk 1]
```diff
@@ -572,11 +571,11 @@ mod prim_array { }
 /// points to:
 ///
 /// ```
-/// let x = &mut [1, 2, 3];
+/// let x = [1, 2, 3];
+/// let x = &mut x[..]; // take a full sclice of `x`
```
Comment (varkor) [inline src/libstd/primitive_docs.rs]:
```suggestion
/// let x = &mut x[..]; // take a full slice of `x`
```

File: src/libstd/primitive_docs.rs [hunk 2]
```diff
@@ -572,11 +571,11 @@ mod prim_array { }
 /// points to:
 ///
 /// ```
-/// let x = &mut [1, 2, 3];
+/// let x = [1, 2, 3];
+/// let x = &mut x[..]; // take a full slice of `x`
```
Comment (Centril) [inline src/libstd/primitive_docs.rs]:
```suggestion
/// let x = &mut x[..]; // Take a full slice of `x`.
```

**Comment by KodrAus** [comment]
Thanks @steveklabnik!

@bors r+ rollup

**Comment by pietroalbini** [comment]
@bors retry
AppVeyor... what's wrong with you today?

**Pull request merged**